### ISSUE TITLE
Support file attachments (uploads)

### ISSUE THREAD

**Issue by alexolog** [EionRobb/purple-discord#164]
As the title says.  The current plugin does not support attaching files.

**Comment by bobbybee** [comment]
Any reason you would actually *want* this, as opposed to uploading it
somewhere else and linking (possibly better for privacy as well,
depending how DiscordBot behaves)?

**Comment by DuendeInexistente** [comment]
One can't really denny that just choosing the file and uploading it in-chat
is more practical, for people who doesn't have some shell script to upload
from the terminal or file browser.

**Comment by bobbybee** [comment]
Sure, I can: a Pidgin plugin could intercept files and instead upload to
a selected host, substituting in a link at will.

---

Hey, Ferb, I knew what we're going to do today!

**Comment by alexolog** [comment]
It's a question of convenience.

I can drag a file into the Discord client (or select it as an attachment) and it will upload it to the Discord servers and post a link.  It works pretty well for channels, if the file size is below the limit, and for private conversations, if the files are not treated as actually private.

I guess the most flexible solution would be to support it with an option to specify a different upload host to upload to, as @bobbybee mentioned above.

And yes, it can be done manually, but it is less convenient.

**Comment by bobbybee** [comment]
> I guess the most flexible solution would be to support it with an
> option to specify a different upload host to upload to, as @bobbybee
> mentioned above.

More specifically, that this should be handled in Pidgin (or whatever
people use) rather than in this prpl. I'm not even sure the prpl *could*
implement this in a convenient generic way...

**Comment by alexolog** [comment]
What about a non-generic way?

**Comment by bobbybee** [comment]
I haven't played with Pidgin/libpurple stuff in a little while; you'd
have to ask Eion about that.

**Comment by DuendeInexistente** [comment]
Maybe you could execute a script and redirect its output to the message
box, having a default host be a simple one like catbox.moe which supports
curl upload.

**Comment by DuendeInexistente** [comment]
Oh and, maybe this'd be better as a standalone plugin, I don't see why have
it only work on Discord.

**Comment by bobbybee** [comment]
> Oh and, maybe this'd be better as a standalone plugin, I don't see why have
> it only work on Discord.

^^ Exactly. I'm not opposed at all to the idea -- I'd just like it to
work on XMPP too. No changes required on our end! :)

**Comment by Michal-Szczepaniak** [comment]
@EionRobb thats good :) I would help if i could

**Comment by TomDataworks** [comment]
I agree any kind of general upload-paste-link feature should be a plugin which works with all protocols - file transfers aren't supported by a lot of the plugins as it is.

**Comment by alexolog** [comment]
Can this please be looked at?
Thank you!

**Comment by Slider-Whistle** [comment]
Images uploaded to any host may be taken down without any notice (remember the photobucket catastrophe?). Theoretically, images uploaded to Discord should stay up for as long as Discord is up. Discord might also one day go down without any notice, but at least everyone should start moving to a different protocol instead.
Also, as far as I know, Pidgin doesn't have any link preview plugins that will embed images.

**Comment by DuendeInexistente** [comment]
> Also, as far as I know, Pidgin doesn't have any link preview plugins that will embed images.

Kiiiind of, there's [an ancient plugin](https://code.google.com/archive/p/pidgin-embeddedvideo/) that would embed videos and allegedly images. But I don't remember it working anymore and the outdated dependencies make it hard to compile and, frankly, are excesive. Why do I need webkit's huge dev packages when pidgin can embed simple media on its own via html tags?

**Comment by Jaegrqualm** [comment]
> Also, as far as I know, Pidgin doesn't have any link preview plugins that will embed images.
Not sure what it is, but there's something currently in use to allow for custom emojis to show up appropriately.

**Comment by T1mL3arn** [comment]
> Pidgin doesn't have any link preview plugins that will embed images.

Google Hangouts plugin has. I don't use the plugin now (and don't know how exactly it works) but I could see preview images that was sent to me and also I could easily drop an image to chat window and send it to someone.

**Comment by MangaD** [comment]
In the official Discord client, I use the feature of uploading images from the clipboard a lot (just crl+v on the text input). It would be wonderful to have it working on this plugin.
